packages/jupyter-ai/src/status.ts

```diff
@@ -0,0 +1,31 @@
+import { IJupyternautStatus } from './tokens';
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+import { IStatusBar } from '@jupyterlab/statusbar';
+import { JupyternautStatus } from './components/statusbar-item';
+
+export const jupyternautStatus: JupyterFrontEndPlugin<IJupyternautStatus> = {
+  id: 'jupyter_ai:jupyternaut-status',
+  description: 'Adds a status indicator for jupyternaut.',
+  autoStart: true,
+  optional: [IStatusBar],
```

Comment:
I was thinking long term here potentially too long term: if we were to support Notebook, we would want to have this plugin work even if the statusbar is absent, but put the indicator elsewhere. Myself putting it in optionals was only to mark this future intent (and hence it is named "status" not "statusbar item") - let me know if you want me to make it required for now though.

Technically, there is no reason to make statusbar optional because the use of the `IJupyternautStatus` in `inlineCompletionProvider` is optional. However, if `IJupyternautStatus` was required by any other plugin then yes lack of the status bar would prevent it from starting.